### PR TITLE
New Package: EBMS

### PR DESCRIPTION
**Pull request by pkondamudi**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ebms/package.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+       similar miniapp provided by Andrew Siegel, whose algorithm is
+       described in [1], where only one process in a compute node
+       is used, and the compute nodes are divided into memory nodes
+       and tracking nodes.    Memory nodes do not participate in particle
+       tracking. Obviously, there is a lot of resource waste in this design."""
+
+    homepage = "http://www.example.com"
+    url = "https://github.com/ANL-CESAR/EBMS/archive/master.tar.gz"
+
+    tags = ['proxy-app']
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ebms/package.py]: i don't think spack has any tags?

Comment (pkondamudi) [inline var/spack/repos/builtin/packages/ebms/package.py]: We were advised by @tgamblin for full-text search purposes.

Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ebms/package.py]: Spack doesn't currently have any tags. I'm wary of adding things to packages that don't do anything. Maybe we should add support for tags and _then_ add tags to packages?

Comment (pkondamudi) [inline var/spack/repos/builtin/packages/ebms/package.py]: Below package has been approved with tags.

https://github.com/LLNL/spack/pull/4683/files

Comment (davydden) [inline var/spack/repos/builtin/packages/ebms/package.py]: I agree, we should not add unsupported fields. If something was merged with this, IMO it's a bug.

Once feature is implemented, we can slowly adjust all packages

Comment (pkondamudi) [inline var/spack/repos/builtin/packages/ebms/package.py]: OK Sure. Will update.

File: var/spack/repos/builtin/packages/ebms/package.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+class Ebms(MakefilePackage):
+    """This is a miniapp for the Energy Banding Monte Carlo (EBMC)
+       neutron transportation simulation code.  It is adapted from a
+       similar miniapp provided by Andrew Siegel, whose algorithm is
+       described in [1], where only one process in a compute node
+       is used, and the compute nodes are divided into memory nodes
+       and tracking nodes.    Memory nodes do not participate in particle
+       tracking. Obviously, there is a lot of resource waste in this design."""
+
+    homepage = "http://www.example.com"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ebms/package.py]: Needs a homepage.

File: var/spack/repos/builtin/packages/ebms/package.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    url = "https://github.com/ANL-CESAR/EBMS/archive/master.tar.gz"
+
+    version('master', git='https://github.com/ANL-CESAR/EBMS.git')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/ebms/package.py]: has to be `develop`

**Pull request merged**